distributed/worker.py

```diff
@@ -1230,12 +1231,12 @@ async def heartbeat(self):
             self.bandwidth_types.clear()
         except CommClosedError:
             logger.warning("Heartbeat to scheduler failed", exc_info=True)
-            await self.close(report=False)
+            await self.close()
         except OSError as e:
             # Scheduler is gone. Respect distributed.comm.timeouts.connect
             if "Timed out trying to connect" in str(e):
                 logger.info("Timed out while trying to connect during heartbeat")
-                await self.close(report=False)
+                await self.close()
```

Comment:
```suggestion
                await self.terminate()
```
⚠️ If the scheduler is gone, but there's a Nanny, I believe this will make the Nanny restart the worker in an infinite loop?

Reply:
Okay, it's not an infinite restart loop (just because `Worker._register_with_scheduler` will actually never give up trying to connect to the scheduler), but it does mean the worker will restart if the scheduler crashes and wait forever for it to show up.

In 3 terminals, run:
```bash
dask-scheduler & echo $!
```
then
```bash
DASK_DISTRIBUTED__COMM__TIMEOUTS__CONNECT="1s" dask-worker localhost:8786 --nanny
```
then
```
sudo kill -9 <scheduler PID printed in first terminal>
```
The scheduler will die without sending the `terminate` signal to workers. The worker will close, the nanny will restart it, and it will then loop forever trying to connect to the nonexistent scheduler.

Is this acceptable? You could argue that if the scheduler crashes, maybe something should bring a new one back up, so it makes sense to restart. But that's assuming a lot about your deployment system, and it should probably be the deployment system's job to do that restarting. To me this is a regression that could lead to wasted resources for anyone counting on the current behavior if the scheduler crashes.